src/components/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,40 @@
+import { Box, IconButton, Button, Divider, SwipeableDrawer } from '@mui/material';
+import ChevronRightIcon from '@mui/icons-material/ChevronRight';
+import DrawerHeader from './DrawerHeader';
+import Logo from '../Logo/Logo';
+import List from '../List/List';
+import { toggleDrawer } from './utils';
+import { DrawerT } from '../../types/types';
+
+const Drawer = ({ open, setOpen, navigate, siteTreeOptions }: DrawerT) => {
+  return (
+    <SwipeableDrawer
+      anchor={'right'}
+      open={open}
+      onClose={toggleDrawer(false, setOpen)}
```

Comment:
I wonder if it would be nicer to define the callback in the `Navigation` component and pass down just the callback. That way it wouldn't be needed to pass down the `setOpen` and also there would be no need to have a util function that returns a function?

What do you think?

Reply:
You are correct. I will define the callback in the `Navigation` component and pass this callback.